File: test/data/src/cs/UpdateLeafFlag.cs
if (phase >= 1 && phase< 4)
{
    if (leafNumber > 0)
    {
        if (hasFlagLeafLiguleAppeared == 0 && (finalLeafNumber > 0 && leafNumber >= finalLeafNumber))
        {
            hasFlagLeafLiguleAppeared = 1;
            if  (calendarMoments.Contains("FlagLeafLiguleJustVisible")==false)
            {
                calendarMoments.Add("FlagLeafLiguleJustVisible");
                calendarCumuls.Add(cumulTT);
                calendarDates.Add(currentdate);
            }
        }
    }
    else
        hasFlagLeafLiguleAppeared = 0;
}


    

File: test/data/phenology/crop2ml/algo/f90/UpdateLeafFlag.f90

    !use crop2mlModules
        
        IF ((phase >= 1) .AND. (phase< 4)) THEN
            IF (leafNumber > 0) THEN
                IF ((hasFlagLeafLiguleAppeared == 0) .AND. (finalLeafNumber > 0) .AND. (leafNumber >= finalLeafNumber)) THEN
                    hasFlagLeafLiguleAppeared = 1;
                    IF  (ALL(calendarMoments/="FlagLeafLiguleJustVisible")) THEN
                        CALL AddToListChar(calendarMoments,"FlagLeafLiguleJustVisible")
                        CALL AddToList(calendarCumuls,cumulTT)
                        CALL AddToListChar(calendarDates, currentdate)
                    END IF
                END IF
            ELSE
                hasFlagLeafLiguleAppeared = 0
            END IF
        END IF


